packages/cli/src/commands/trigger.ts

```diff
@@ -0,0 +1,129 @@
+import { Flags } from '@oclif/core'
+
+import * as api from '../rest/api'
+import { AuthCommand } from './authCommand'
+import { loadChecklyConfig } from '../services/checkly-config-loader'
+import { splitConfigFilePath } from '../services/util'
+import type { Region } from '..'
+import CheckRunner, { RunLocation, Events, PrivateRunLocation } from '../services/check-trigger-runner'
+import config from '../services/config'
+
+const DEFAULT_REGION = 'eu-central-1'
+
+export default class Trigger extends AuthCommand {
+  static hidden = false
+  static description = 'Trigger checks on Checkly'
+  static flags = {
+    location: Flags.string({
+      char: 'l',
+      description: 'The location to run the checks at.',
+    }),
+    'private-location': Flags.string({
+      description: 'The private location to run checks at.',
+      exclusive: ['location'],
+    }),
+    tag: Flags.string({
+      char: 't',
+      description: 'Run checks matching the specified tag.',
+      multiple: true,
+      required: true,
+    }),
+    config: Flags.string({
+      char: 'c',
+      description: 'The Checkly CLI config filename.',
+    }),
+  }
+
+  async run (): Promise<void> {
+    const { flags } = await this.parse(Trigger)
+    const {
+      location: runLocation,
+      'private-location': privateRunLocation,
+      config: configFilename,
+      tag: targetTags,
+    } = flags
+    const { configDirectory, configFilenames } = splitConfigFilePath(configFilename)
+    
+    // TODO: We don't _realy_ need the checkly config file. We should handle it missing gracefully.
+    const { config: checklyConfig } = await loadChecklyConfig(configDirectory, configFilenames)
+    const location = await this.prepareRunLocation(checklyConfig.cli, {
+      runLocation: runLocation as keyof Region,
+      privateRunLocation,
+    })
+    
+    const runner = new CheckRunner(
+      config.getAccountId(),
+      config.getApiKey(),
+      targetTags,
+      location,
+      240,
+      false,
+    )
+    // TODO: the reporter is quite coupled to having logical Ids. Need to work on that.
+    // let reporter
+    runner.on(Events.RUN_STARTED, (checks) => {
```

Comment:
I didn't get it working, but we should be able to set up the reporter here in the `RUN_STARTED` event. We won't know the checks that will be run beforehand, but in `RUN_STARTED` we receive a list of checks.